internal/controllers/generic/conditions.go

```diff
@@ -32,27 +32,21 @@ type WithConditionsApplyConfiguration[T any] interface {
 	WithConditions(...*applyconfigv1.ConditionApplyConfiguration) T
 }
 
-func SetCommonConditions[T any](orcObject orcv1alpha1.ObjectWithConditions, applyConfig WithConditionsApplyConfiguration[T], isAvailable, isUpToDate bool, progressMessage *string, err error, now metav1.Time) {
+func SetCommonConditions[T any](orcObject orcv1alpha1.ObjectWithConditions, applyConfig WithConditionsApplyConfiguration[T], isAvailable bool, progressStatus []ProgressStatus, err error, now metav1.Time) {
 	availableCondition := applyconfigv1.Condition().
 		WithType(orcv1alpha1.ConditionAvailable).
 		WithObservedGeneration(orcObject.GetGeneration())
 	progressingCondition := applyconfigv1.Condition().
 		WithType(orcv1alpha1.ConditionProgressing).
 		WithObservedGeneration(orcObject.GetGeneration())
 
-	if err == nil {
-		if isUpToDate {
-			progressingCondition.
-				WithStatus(metav1.ConditionFalse).
-				WithReason(orcv1alpha1.ConditionReasonSuccess).
-				WithMessage("OpenStack resource is up to date")
-		} else {
-			progressingCondition.
-				WithStatus(metav1.ConditionTrue).
-				WithReason(orcv1alpha1.ConditionReasonProgressing).
-				WithMessage(ptr.Deref(progressMessage, "Reconciliation is progressing"))
-		}
-	} else {
+	// We are Progressing iff we are anticipating being reconciled again. This
```

Comment:
just a small typo
```suggestion
	// We are Progressing if we are anticipating being reconciled again. This
```

Reply:
That's not a typo ;)

https://en.wikipedia.org/wiki/If_and_only_if